pkg/asset/machines/worker.go

```diff
@@ -399,7 +399,11 @@ func (w *Worker) Generate(ctx context.Context, dependencies asset.Parents) error
 			}
 
 			if mpool.InstanceType == "" {
-				instanceTypes := awsdefaults.InstanceTypes(installConfig.Config.Platform.AWS.Region, installConfig.Config.ControlPlane.Architecture, configv1.HighlyAvailableTopologyMode)
+				arch := installConfig.Config.ControlPlane.Architecture
```

Comment:
this whole loop should change to: 

```
for i, pool := range ic.Compute {
```
and something like
```
osImage := strings.SplitN(string(*rhcosImage.ComputeImage[i]), ",", 2)
```

Reply:
No need, see comment above about there being only one compute machine pool.